2-mandatory/3-financial-times.js

```diff
@@ -14,7 +17,22 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+  // let shortestTitleLength = Infinity;
+  // return allArticleTitles.reduce((acc, title) => {
+  //     if (title.split("").length < shortestTitleLength) {
+  //       shortestTitleLength = titleLengthInWords;
+  //       return title;
+  //     } else {
+  //       return acc;
+  //     }
+  // }, "");
+  let shortestTitle = allArticleTitles[0];
+  for (let i = 1; i < allArticleTitles.length; i++) {
+    if (allArticleTitles[i].split("").length < shortestTitle.split("").length) {
```

Comment:
Thanks, Ellie. The first one takes each character as one item of the array. The second one takes the words as an item of the array. I think I should have used the second one.Because I need to count the words not the characters.

Reply:
Hi Ellie, the first one takes all the characters of the array as one item, therefore the length of the array is equal to the number of the characters including spaces. The second one is taking all the words in the array as an item, therefore the number of the words is the length of the array. I changed mine to the second one.